cli/operator/node.go

```diff
@@ -402,6 +421,38 @@ func setupDB(logger *zap.Logger, eth2Network beaconprotocol.Network) (*kv.Badger
 	return db, nil
 }
 
+func setupSlashingProtectionDB(ctx context.Context, logger *zap.Logger, network beaconprotocol.BeaconNetwork) (*kv.BadgerDB, error) {
+	// Validate that provided slashing protection DB path directory exists
+	if _, err := os.Stat(cfg.SlashingProtectionOptions.DBPath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("slashing protection database does not exist, please create it using the 'create-slashing-protection-db' command: %w", err)
+	}
+
+	// Validate that the slashing protection DB and node DB are not in the same directory
+	if filepath.Dir(cfg.DBOptions.Path) == filepath.Dir(cfg.SlashingProtectionOptions.DBPath) {
+		return nil, fmt.Errorf("slashing protection database and node database cannot be in the same directory")
+	}
+
+	options := basedb.Options{
+		Path:       cfg.SlashingProtectionOptions.DBPath,
+		SyncWrites: true,
+	}
+
+	db, err := kv.New(ctx, logger, options)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open slashing protection DB: %w", err)
+	}
+
+	// If the sp db was created with cli command, it should include "genesis" version of the db
+	// this way we can validate the db was created via the cli command. if not, we should fail
+	storage := ekm.NewSlashingProtectionStorage(db, logger, []byte(network.GetBeaconNetwork()))
+	value, found, err := storage.GetVersion()
+	if err != nil || !found || value != ekm.GenesisVersion {
+		return nil, fmt.Errorf("slashing protection database was not created via cli command, please create it using the 'create-slashing-protection-db' command: %w", err)
```

Comment:
I think a separate error for cases if `err` is `nil` would look better and would help understand what exactly is wrong